### ISSUE TITLE
[ux] style copy-link modal

### ISSUE THREAD

**Issue by yoshuawuyts** [datproject/dat-desktop#103]
different folks can do different parts, but I think this is what needs to happen to style the modal a bit further

### todo
- [x] stop copying link by default
- [x] build out a little model for the modal
- [x] style elements
- [x] fix regression with weird visual background things happening

---

### screenshots
![](https://cloud.githubusercontent.com/assets/53684/20926558/0c59dcd0-bc11-11e6-85bc-18be18478895.png)
![](https://cloud.githubusercontent.com/assets/53684/20926562/153ee8cc-bc11-11e6-9223-ba09d64b13e4.png)

**Comment by Kriesse** [comment]
@yoshuawuyts nice! styling work started here: https://github.com/datproject/dat-desktop/tree/dat-link-modal

**Comment by Kriesse** [comment]
@yoshuawuyts Can you make it so that when the link is copied to the clipboard, the class `show-confirmation` is added here? https://github.com/datproject/dat-desktop/blob/master/elements/modal.js#L170

**Comment by yoshuawuyts** [comment]
@Kriesse done! - airport wifi is blocking `git push` tho (no UDP connections) so will push when home :D - really liking how it's shaping up :sparkles:

<img width="1280" alt="screen shot 2016-12-08 at 17 26 52" src="https://cloud.githubusercontent.com/assets/2467194/21018022/a3965a82-bd6b-11e6-90bf-72abe5f48cbf.png">

**Comment by yoshuawuyts** [comment]
@Kriesse in terms of CSS code, I noticed we got [about 100 lines of CSS inline](https://github.com/datproject/dat-desktop/blob/master/elements/modal.js#L8-L119) for the modal - would it perhaps be an idea to replace most of it with tachyons before we close this issue?

**Comment by Kriesse** [comment]
@yoshuawuyts I'm still going back and forth trying to come up with a set of rules for when to use a tachyons class vs when to use a custom class. Went through the CSS and cleaned up where it made sense, also removed all scss, please review :) https://github.com/datproject/dat-desktop/pull/110

So far, it turns out certain styles that our UI requires don't come via tachyons. So we have to add them as custom rules.
Example: `border: 1px solid var(--color-neutral-30);`
In the above example, we could of course use tachyons to set `border-width` and `border-style` on the element, and only add a custom rule for `border-color`. 
Another example:
````css
position: relative;
top: -.3rem;
````
Again, the first rule could be set with tachyons, and the second would require a custom class. But I'd find that inconsistent and would really like to avoid it for code readability reasons.

I have some ideas and questions about how to use customized tachyons classes, but need help with implementing it. Let's chat details!

@karissa For reusability, we plan to use https://github.com/datproject/dat-elements. So e.g. the modal would be its own reusable component and come with CSS either way, so no need for an external stylesheet anymore.

**Comment by yoshuawuyts** [comment]
@karissa yup, what @Kriesse said

---

@Kriesse I think if we were to create a classes version of https://www.npmjs.com/package/dat-colors we could go a long way - I feel colors is where we diverge most from tachyons, and fixing that would be good.

Probably the other two points where we differ significantly is our font styles, but we've solved that - and centering icons, but I feel that's also something we can solve.

But yeah like we talked about two days ago: right now we're moving a whole bunch of stuff around so perhaps creating abstractions might not make the most sense. On the other hands perhaps some abstractions (e.g. color classes) might allow us to iterate faster; I'm not entirely sure, but I trust your judgement.

Let's def catch up again soon; feel our chats have been very useful so far :sparkles:
